Formula/r/rustup.rb

```diff
@@ -34,7 +34,16 @@ def install
        rust-gdb rust-gdbgui rust-lldb rustc rustdoc rustfmt rustup].each do |name|
       bin.install_symlink bin/"rustup-init" => name
     end
-    generate_completions_from_executable(bin/"rustup", "completions")
```

Comment:
Why doesn't this generate everything?

Reply:
By default, this command generates only a script for `rustup`. It also supports `cargo`, but it needs to be called separately.

Here’s how it looks in practice:

```
$ rustup completions zsh | grep compdef
#compdef rustup
    compdef _rustup rustup
$ rustup completions zsh cargo | grep compdef
#compdef cargo
```

Reply:
Is there a way to still get the shell from the `$SHELL` variable when you generate cargo completion? Because the current suggestion works around the whole automatic generation and seems much more brittle than just leaving out the cargo completions.

Reply:
Unfortunately, no, the shell depends solely on the third argument. The order of arguments is also fixed, following the pattern `rustup completions <shell> <command>`.

Cargo is Rust’s package manager and build system, so I think having autocompletion for it is a must.

Reply:
> Cargo is Rust’s package manager and build system, so I think having autocompletion for it is a must.

Sure, but at the same time rustup is not rust.

Reply:
That’s right. But I assume that once someone installs rustup, they probably won’t need to install rust separately.

I’m not happy with the current solution either. Alternatively, I could try adding a placeholder for the shell in the `generate_completions_from_executable`, then the code would look like this:

```rb
generate_completions_from_executable(bin/"rustup", "completions")
generate_completions_from_executable(bin/"rustup", "completions", :shell, "cargo", shells: [:bash, :zsh])
```

What do you think about this approach?

Reply:
Yeah, I think that's a better way to do this.